File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    result = []
    binary_arr1 = []
    binary_arr2 = []
    and_result = []
    
    for ar1, ar2 in zip(arr1, arr2):
        binary_arr1 = format(ar1, 'b')
        binary_arr2 = format(ar2, 'b')
        
        and_result = format(int(binary_arr1, 2) | int(binary_arr2, 2), 'b')
        
        result.append(and_result.rjust(n, '0').replace('1', '#').replace('0', ' '))
    
    return result

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i = 0; i < n; i++) {
            // String binary1 = Integer.toBinaryString(arr1[i]);
            // String binary2 = Integer.toBinaryString(arr2[i]);
            String binary1 = String.format("%" + n + "s", Integer.toBinaryString(arr1[i])).replace(' ', '0');
            String binary2 = String.format("%" + n + "s", Integer.toBinaryString(arr2[i])).replace(' ', '0');
            StringBuilder sb = new StringBuilder();
            
            for (int k = 0; k < binary1.length(); k++) {
                if (binary1.charAt(k) == '1' || binary2.charAt(k) == '1') {
                    sb.append("#");
                } else {
                    sb.append(" ");
                }
            }
            
            answer[i] = sb.toString();
        }
        
        return answer;
    }
}